### DISCUSSION TITLE
Add 3D Panel support for geometry_msgs/Polygon messages

### DISCUSSION THREAD

**Discussion by erbornemeier** [foxglove/community#653] [Q&A]
Currently the 3D panel will only display geometry_msgs/PolygonStamped messages. I've got a message definition that is similar to the below example that I would like to view in this panel.
```
Header header

geometry_msgs/Polygon example_polygon
geometry_msgs/Polygon[] example_polygon_list
```
Since these polygons all have an associated timestamp, I think it should be doable to display these the same as PolygonStamped messages.

One suggestion would be to add support for geometry_msgs/Polygon generically by using the receive time instead of a header.

**Comment by jtbandes** [comment]
It seems like there are two requests here. One is adding support for Polygon (we only support PolygonStamped) today. The other would be something like https://github.com/foxglove/community/discussions/228 but for the 3D panel (ability to display supported message types that occur as sub-fields of another message). Is that accurate?

**Comment by defunctzombie** [comment]
FWIW you can use a message converter extension or a user script to make your custom message renderable as a PolygonStamped or some other geometry. These techniques are flexible to any kind of nesting or custom messages.
